core/src/main/java/org/eclipse/hono/util/TimeUntilDisconnectNotification.java

```diff
@@ -187,4 +185,54 @@ public long getMillisecondsUntilExpiry() {
             return (milliseconds > 0 ? milliseconds : 0);
         }
     }
+
+    /**
+     * Checks if a device is currently connected to a protocol adapter.
+     * <p>
+     * If this method returns {@code true} an attempt could be made to send a command to the device.
+     * <p>
+     * This method uses the message's creation time and TTD value to determine the point in time
+     * until which the device will remain connected.
+     *
+     * @param ttd The TTD value.
+     * @param creationTime The creation time of the message. If {@code null} the device is considered as disconnected.
+     * @return {@code true} if the TTD value contained in the message indicates that the device will
+     *         stay connected for some additional time.
+     */
+    public static boolean isDeviceCurrentlyConnected(final Integer ttd, final Long creationTime) {
+
+        return Optional.ofNullable(ttd).map(ttdValue -> {
+            if (ttdValue == MessageHelper.TTD_VALUE_UNLIMITED) {
```

Comment:
Hmmm, same problem, I guess. If we really want to get rid of the AMQP dependencies we will probably need to move the field constants to some more technology-agnostic class, e.g. *Constants*. But I guess we should do that in another PR ...

Reply:
I think, if we really want to get rid of unnecessary dependencies of the client module we need to cut the Maven dependency to hono-core which is still AMQP-specific.

Reply:
If we only use classes from hono-core which do not depend on e.g. vertx-proton and/or proton-j then we can at least exclude those artifacts from the declared dependency to hono-core. But that is not possible if we need e.g. MessageHelper and MessageHelper uses proton-j ...